src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java

```diff
@@ -0,0 +1,295 @@
+                : this.contentType + "; charset=" + encoding;
+        if (!StringUtils.isEmpty(content)) {
+            this.content = content;
+        }
+        this.contentTransferEncoding = StringUtils.isEmpty(contentTransferEncoding)
+                ? EmailConstants.DEFAULT_CONTENT_TRANSFER_ENCODING : contentTransferEncoding;
+        return this;
+    }
+
+    /**
+     * Adds attachments to the email
+     *
+     * @param attachments file paths of the attachments to be added to the email
+     * @return MessageBuilder instance
+     */
+    public MessageBuilder withAttachments(String attachments) {
+
+        if (!StringUtils.isEmpty(attachments)) {
+            this.attachments = attachments;
+        }
+        return this;
+    }
+
+    /**
+     * Build MIME Message using configured parameters
+     *
+     * @return MIME Message with the configured values
+     * @throws MessagingException if failed to build message
+     */
+    public MimeMessage build() throws MessagingException, IOException {
+
+        if (attachments != null && !attachments.isEmpty()) {
+            MimeMultipart multipart = new MimeMultipart();
+            MimeBodyPart body = new MimeBodyPart();
+            setMessageContent(body);
+            multipart.addBodyPart(body);
+            String[] attachFiles = attachments.split(",");
+            for (String filePath : attachFiles) {
+                addAttachment(multipart, filePath);
+            }
+            message.setContent(multipart, MULTIPART_TYPE);
+        } else {
+            setMessageContent(message);
+        }
+        return message;
+    }
+
+    /**
+     * Sets message content in message body
+     *
+     * @param part MimePart the content should be set to
+     * @throws MessagingException if an error occurs while setting the content
+     */
+    private void setMessageContent(MimePart part) throws MessagingException {
+
+        part.setDisposition(INLINE);
+        part.setContent(content, contentType);
+        part.setHeader(CONTENT_TYPE_HEADER, contentType);
+        if (!StringUtils.isEmpty(contentTransferEncoding)) {
+            part.setHeader(CONTENT_TRANSFER_ENCODING_HEADER, contentTransferEncoding);
+        }
+    }
+
+    /**
+     * Add attachment to message
+     *
+     * @param multipart Multi part body the messages should be added to
+     * @param filePath  File path of the attachment
+     * @throws MessagingException if failed to set attachments
+     * @throws IOException        if an error occurred while reading attachment content
+     */
+    private void addAttachment(MimeMultipart multipart, String filePath) throws MessagingException, IOException {
+
+        MimeBodyPart part = new MimeBodyPart();
+        File file = new File(filePath);
+        try (InputStream fin = new FileInputStream(file)) {
+            part.setDisposition(ATTACHMENT);
+            part.setFileName(file.getName());
+            Tika tika = new Tika();
```

Comment:
Shall we check if this works  with other platforms as well - like Windows 10 and Linux?